dvc/repo/add.py

```diff
@@ -2,6 +2,22 @@
 from dvc.stage import Stage
 
 
+def cleanup_gitignore(add_method):
```

Comment:
Let's move this to `dvc/scm`. Also, since we do have mercurial support upcoming, maybe we could name it something not git-specific?